### ISSUE TITLE
[bug] XcodeDeps does not correctly handle transitive components

### ISSUE THREAD

**Issue by DoDoENT** [conan-io/conan#14887]
### Environment details

* Operating System+version: macos ventura
* Compiler+version: 14.0.3
* Conan version: 2.0.13
* Python version: 3.11

### Steps to reproduce

Let's have 3 simple packages:

The first package, defines two components:

```python
from conan import ConanFile


class PkgWithComponents(ConanFile):
    name = 'has_components'
    version = '1.0'
    settings = 'os', 'compiler', 'arch', 'build_type'

    def package_info(self):
        self.cpp_info.components['first'].libs = ['first']
        self.cpp_info.components['second'].libs = ['donottouch']
        self.cpp_info.components['second'].requires = ['first']
```

The second package depends on first, but uses only first component from it:

```python
from conan import ConanFile


class PkgUsesComponent(ConanFile):
    name = 'uses_components'
    version = '1.0'
    settings = 'os', 'compiler', 'arch', 'build_type'

    def requirements(self):
        self.requires('has_components/1.0')

    def package_info(self):
        self.cpp_info.libs = ['uses_only_first']
        self.cpp_info.requires = ['has_components::first']
```

The third is a consumer package for iOS app, which uses the second package:

```python
from conan import ConanFile
from conan.tools.apple import XcodeDeps, XcodeToolchain


class TheApp(ConanFile):
    requires = 'uses_components/1.0'
    settings = 'os', 'compiler', 'arch', 'build_type'

    def generate(self):
        tc = XcodeToolchain(self)
        tc.generate()

        deps = XcodeDeps(self)
        deps.generate()

    def layout(self):
        self.folders.generators = f'conan-data/{self.settings.build_type}/{self.settings.os.sdk}'
```

Export first and second package into the local cache, and run `conan install` for the third package.

The resulting `conandeps.xcconfig` looks like this:

```
// Conan XcodeDeps generated file
// Includes all direct dependencies

#include "conan_has_components.xcconfig"

#include "conan_uses_components.xcconfig"
```

which is wrong. Namely, `has_components` is not a direct dependency, so it should not be mentioned here. The problematic part is that `conan_has_components.xcconfig` includes both `conan_has_components_first.xcconfig` and `conan_has_components_second.xcconfig`, meaning that xcode variables get updated as if the second component is used, even though it isn't (`-ldonottouch` appears in final link flags).

The `CMakeDeps` works correctly for this scenario - the `uses_components::uses_components` correctly links only to `has_components::first` and `-ldonottouch` does not appear in the final link flags.

**Comment by czoido** [comment]
Hi @DoDoENT,

Thanks a lot for reporting and for the detailed instructions on how to reproduce that. Indeed this seems like a bug adding extra transitive dependencies that should not be there. I'll mark this for next release to be fixed.

**Comment by DoDoENT** [comment]
No problem.

The workaround for the bug is to simply manually generate the `conandeps.xcconfig`, something like this:

```python
    def generate(self):
        toolchain = XcodeToolchain(self)
        toolchain.generate()

        deps = XcodeDeps(self)
        deps.generate()

        # workaround for https://github.com/conan-io/conan/issues/14887
        correct_conandeps = textwrap.dedent("""
            #include "conan_uses_components.xcconfig"
        """)
        save(self, "conandeps.xcconfig", correct_conandeps)
```

**Comment by czoido** [comment]
Hi @DoDoENT, 
I'm proposing a fix in https://github.com/conan-io/conan/pull/14898. Would it be possible for you to test those changes to see if the fix solves your problem?
Thanks a lot.

**Comment by DoDoENT** [comment]
Thank you! I just tested this on my real-world project and it works correctly!

**Comment by czoido** [comment]
@DoDoENT we have just merged the fix, it will be in next 2.0.14. Thanks a lot for reporting
